### ISSUE TITLE
[FR] "Native Function result unused" inspection

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1146]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Feature request                          |
| Plugin         | Php Inspections (EA Extended) |

Current behaviour (description/screenshot:)
--
There is  a PhpStorm inspection called  "Expression result unused". It detects when some simple expression are unused, for example:
```php
'a' == 'b'; // Expression result unused
```
It is useful in finding dead code but I believe we could do better by looking at the result of native functions of PHP.

Expected behaviour
--
There are a lot of native functions in PHP. Some of them have a use on their own (for example mkdir), some have optional references and can do something even when their result is not used (&$matches for preg_match) but the great majority of them are useless when their result is unused.

It could be great if those former functions (and possibly the ones with optional references when not used) could be displayed as "Function result unused". For example:

```php
count($array); // Function result unused
str_replace('foo', 'bar', 'foobar'); // Function result unused
str_replace('foo', 'bar', 'foobar', $count); // false positive: $count has a reference and could be useful on its own
ob_start(); // false positive: do something on its own
```

I can try to provide a list of common functions who do nothing on their own and a list of functions with optional references if needed.

Any feedback?

**Comment by kalessil** [comment]
'Unused function result' inspection already exists in Php Inspections (EA Ultimate) - the commercial version of the analyzer. Hence marking as a duplicate.
